creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -772,6 +775,48 @@ class SnapbackSM {
       nodeUser.user_id % ModuloBase === this.currentModuloSlice
     )
   }
+
+  /**
+   *
+   */
+  async secondaryPromise (secondaryUrl, wallet, primaryClockVal, timeoutMs) {
+    const start = Date.now()
+
+    while (Date.now() - start < timeoutMs) {
+      try {
+        const secondaryClockStatusResp = await axios({
+          method: 'get',
+          baseURL: secondaryUrl,
+          url: `/users/clock_status/${wallet}`,
+          responseType: 'json',
+          timeout: 1000 // 1000ms = 1s
+        })
+        const { clockValue: secondaryClockVal } = secondaryClockStatusResp.data.data
+
+        // If secondary is synced, return successfully
+        if (secondaryClockVal >= primaryClockVal) {
+          return
+        }
+
+        // Send syncRequest to secondary
+        await this.enqueueSync({
```

Comment:
this could potentially be moved up above the while. we really only need the while for polling. also right now doesn't it make an unnecessary call to the secondary to fetch clock status before enqueueing the sync?

Reply:
fair enough, changed